File: examples/apps/react-native/src/features/auth/data/AuthRepositoryImpl.ts
import { ApiClient } from '../../../core/network/ApiClient';
import { AuthSession, User } from '../domain/entities/User';
import { AuthRepository } from '../domain/repositories/AuthRepository';

export class AuthRepositoryImpl implements AuthRepository {
  constructor(private readonly client: ApiClient) {}

  async login(email: string, password: string) {
    const session = await this.client.post<AuthSession>('/api/auth/login', {
      email,
      password,
    });
    this.client.setToken(session.token);
    return session;
  }

  async register(name: string, email: string, password: string) {
    const session = await this.client.post<AuthSession>('/api/auth/register', {
      name,
      email,
      password,
    });
    this.client.setToken(session.token);
    return session;
  }

  updateProfile(name: string, email: string, phone: string) {
    return this.client.put<User>('/api/profile', { name, email, phone });
  }
}

File: examples/apps/flutter/lib/features/auth/data/repositories/auth_repository_impl.dart
import 'package:app_prueba_mobilelab/features/auth/data/datasources/auth_remote_data_source.dart';
import 'package:app_prueba_mobilelab/features/auth/domain/entities/app_user.dart';
import 'package:app_prueba_mobilelab/features/auth/domain/repositories/auth_repository.dart';

class AuthRepositoryImpl implements AuthRepository {
  const AuthRepositoryImpl(this._remote);

  final AuthRemoteDataSource _remote;

  @override
  Future<AuthSession> login({required String email, required String password}) {
    return _remote.login(email: email, password: password);
  }

  @override
  Future<AuthSession> register({
    required String name,
    required String email,
    required String password,
  }) {
    return _remote.register(name: name, email: email, password: password);
  }

  @override
  Future<AppUser> getProfile() => _remote.getProfile();

  @override
  Future<AppUser> updateProfile({
    required String name,
    required String email,
    required String phone,
  }) {
    return _remote.updateProfile(name: name, email: email, phone: phone);
  }
}
